### PR TITLE
Add support for mTLS enabled clusters in WorkerProcess

### LINKED ISSUE #214: Add support to mTLS on worker process

### PR DESCRIPTION
**Pull request by alexandrevilain**
This PR closes #214

### CODE REVIEW

File: api/v1beta1/temporalworkerprocess_defaults.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+package v1beta1
+
+import v1 "k8s.io/api/core/v1"
+
+// Default set default fields values.
+func (w *TemporalWorkerProcess) Default() {
+	if w.Spec.Builder != nil {
```
Comment (ktenzer) [inline api/v1beta1/temporalworkerprocess_defaults.go:24]: if worker.Spec.Builder.BuilderEnabled()

Comment (ktenzer) [inline api/v1beta1/temporalworkerprocess_defaults.go:24]: if !worker.Spec.Builder.BuilderEnabled() {

**Review by ktenzer — CHANGES_REQUESTED** [review]

File: api/v1beta1/temporalworkerprocess_defaults.go (hunk 1 above)
Comment (ktenzer) [reply api/v1beta1/temporalworkerprocess_defaults.go:24]: I would use the bool as then regardless of what gets set in CR if builderEnabled is false we wont validate those settings

Comment (rawkode) [reply api/v1beta1/temporalworkerprocess_defaults.go:24]: I wonder if this would fix #233 ?

Comment (alexandrevilain) [reply api/v1beta1/temporalworkerprocess_defaults.go:24]: @rawkode you're right ;)

File: controllers/temporalcluster_controller.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ type TemporalClusterReconciler struct {
 //+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;delete
 //+kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;delete
 //+kubebuilder:rbac:groups="networking.k8s.io",resources=ingresses,verbs=get;list;watch;create;update;delete
-//+kubebuilder:rbac:groups="cert-manager.io",resources=certificates;issuers,verbs=get;list;watch;create;update;delete
+//+kubebuilder:rbac:groups="cert-manager.io",resources=certificates;issuers;clusterissuers,verbs=get;list;watch;create;update;delete
```
Comment (alexandrevilain) [inline controllers/temporalcluster_controller.go:69]: To remove

File: examples/cluster-mtls/02-temporal-cluster.yaml [hunk 3]
```diff
@@ -40,12 +40,14 @@ spec:
       clientCertificates: 1h
       frontendCertificate: 1h
       internodeCertificate: 1h
+    issuersConfiguration:
```
Comment (alexandrevilain) [inline examples/cluster-mtls/02-temporal-cluster.yaml:43]: To remove

**Comment by alexandrevilain** [comment]
This PR fixes #197

**Pull request merged**